### PR TITLE
adding [clearSearchOnSelect] prop

### PR DESCRIPTION
**Pull request by TimasJackus**
Contributing to #1088 issue, allowing to prevent clear search on item select:

- Added `clearSearchOnSelect` input with default value `true`.
- It applies only to single select.

### CODE REVIEW

File: src/ng-select/lib/ng-select.component.ts [hunk 1]
```diff
@@ -440,6 +441,8 @@ export class NgSelectComponent implements OnDestroy, OnChanges, AfterViewInit, C
         if (this.closeOnSelect || this.itemsList.noItemsToSelect) {
             this.close();
         }
+
+        this._preventClearSearch(item);
```
Comment (varnastadeus) [inline src/ng-select/lib/ng-select.component.ts]: this works only the first time. Next time after closing text is not editable

Comment (varnastadeus) [reply src/ng-select/lib/ng-select.component.ts]: Also after fully clearing input - select value reappears

Comment (varnastadeus) [reply src/ng-select/lib/ng-select.component.ts]: AwesomeScreenshot-2019-7-9-1562681592879

**Comment by TimasJackus** [comment]
@varnastadeus updated.

**Comment by monis001** [comment]
Any update on this ?  when is pipeline to get shipped ?

**Comment by anjmao** [comment]
Hi @TimasJackus , could you fix conflicting files so I can merge it. Sorry for late response.

**Comment by monis001** [comment]
Yeah, looking forward for this 
Thanks

**Comment by TimasJackus** [comment]
resolved

**Comment by varnastadeus** [comment]
@anjmao Implementation doesn't handle all use cases and as I remember it didn't work properly

**Comment by Komock** [comment]
@TimasJackus Any news here? This is a really useful feature

**Comment by TimasJackus** [comment]
@Komock I'll try to fix it on weekend, really busy week.

**Pull request closed without merging**